src/CodeGen_OpenCL_Dev.cpp

```diff
@@ -16,6 +17,131 @@ using std::sort;
 
 static ostringstream nil;
 
+// OpenCL doesn't support vectors of bools, this mutator rewrites IR
+// to use signed integer vectors instead.
+class EliminateBoolVectors : public IRMutator {
```

Comment:
Expand on this comment a little - I think this mutator converts all trees of logical ops on bool vectors to bitwise ops on int vectors, wrapping the comparisons at the leaves in upcasts, and wrapping the use in a select argument in a downcast (by comparing to zero). The actual codegen class then peepholes through the downcast and upcast to use OpenCL's comparisons (which return masks) and OpenCL's select (which takes a mask)